src/model.ts

```diff
@@ -404,6 +406,7 @@ class FileBrowserModel implements IDisposable {
    * Load the list of kernel specs.
    */
   private _getKernelSpecs(): void {
+    this._kernelSpecs = [];
```

Comment:
This will make the specs empty until the session promise fulfills.

What about only assigning to `this._kernelSpecs` with the final sorted results?